src/main/java/com/pinterest/secor/common/SecorSchemaRegistryClient.java

```diff
@@ -0,0 +1,47 @@
+package com.pinterest.secor.common;
+
+import io.confluent.kafka.schemaregistry.client.CachedSchemaRegistryClient;
+import io.confluent.kafka.serializers.KafkaAvroDecoder;
+import org.apache.avro.Schema;
+import org.apache.avro.generic.GenericRecord;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Map;
+import java.util.Properties;
+import java.util.concurrent.ConcurrentHashMap;
+
+public class SecorSchemaRegistryClient {
+
+    private static final Logger LOG = LoggerFactory.getLogger(SecorSchemaRegistryClient.class);
+
+    private KafkaAvroDecoder decoder;
+    private final static Map<String, Schema> schemas = new ConcurrentHashMap<>();
+
+    public SecorSchemaRegistryClient(SecorConfig config) {
+        try {
+            Properties props = new Properties();
+            props.put("schema.registry.url", config.getSchemaRegistryUrl());
+            CachedSchemaRegistryClient schemaRegistryClient = new CachedSchemaRegistryClient(config.getSchemaRegistryUrl(), 30);
+            decoder = new KafkaAvroDecoder(schemaRegistryClient);
+        } catch (Exception e){
+            LOG.error("Error initalizing schema registry", e);
+            throw new RuntimeException(e);
+        }
+    }
+
+    public GenericRecord decodeMessage(String topic, byte[] message) {
+        GenericRecord record = (GenericRecord) decoder.fromBytes(message);
+        Schema schema = record.getSchema();
+        schemas.putIfAbsent(topic, schema);
```

Comment:
Does this mean you will not update the topic schema to latest?

Reply:
That's a good point @HenryCaiHaiying, I will add unit tests to this class for that scenario.

Reply:
So you don't want to update the schema?

Reply:
No, I do want to. I'll make changes and add tests to ensure they are updated. The SchemaRegistryClient should already handle changes, so this wrapper may even be unnecessary.

Reply:
@HenryCaiHaiying I fixed and tested this behavior: please see the new `SecorSchemaRegistryClientTest`